hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieFileSliceReader.java

```diff
@@ -35,24 +35,24 @@
 /**
  * Reads records from base file and merges any updates from log files and provides iterable over all records in the file slice.
  */
-public class HoodieFileSliceReader implements Iterator<HoodieRecord<? extends HoodieRecordPayload>> {
-  private Iterator<HoodieRecord<? extends HoodieRecordPayload>> recordsIterator;
+public class HoodieFileSliceReader<T extends HoodieRecordPayload> implements Iterator<HoodieRecord<T>> {
+  private Iterator<HoodieRecord<T>> recordsIterator;
 
-  public static <R extends IndexedRecord, T extends HoodieRecordPayload> HoodieFileSliceReader getFileSliceReader(
+  public static <R extends IndexedRecord, T> HoodieFileSliceReader getFileSliceReader(
       HoodieFileReader<R> baseFileReader, HoodieMergedLogRecordScanner scanner, Schema schema, String payloadClass,
       Option<Pair<String,String>> simpleKeyGenFieldsOpt) throws IOException {
     Iterator<R> baseIterator = baseFileReader.getRecordIterator(schema);
     while (baseIterator.hasNext()) {
       GenericRecord record = (GenericRecord) baseIterator.next();
-      HoodieRecord<T> hoodieRecord = simpleKeyGenFieldsOpt.isPresent()
+      HoodieRecord<? extends HoodieRecordPayload> hoodieRecord = simpleKeyGenFieldsOpt.isPresent()
```

Comment:
@satishkotha Here's another change which I had to do to accommodate virtual keys changes https://github.com/apache/hudi/pull/3315/files#diff-ec6e010de1137a42412dad69d088607b543c7904e3e5a68e460fc64c3e7868d1